src/fairchem/core/modules/normalizer.py

```diff
@@ -7,50 +7,209 @@
+    ):
+        # check if state dict is legacy state dicts
+        if isinstance(state_dict["mean"], float):
+            state_dict.update({k: torch.tensor(state_dict[k]) for k in ("mean", "std")})
+
+        super().load_state_dict(state_dict, strict=strict, assign=assign)
+
+
+def create_normalizer(
+    file: str | Path | None = None,
+    state_dict: dict | None = None,
+    tensor: torch.Tensor | None = None,
+    mean: float | torch.Tensor | None = None,
+    std: float | torch.Tensor | None = None,
+) -> Normalizer:
+    """Build a target data normalizers with optional atom ref
+
+    Only one of file, state_dict, tensor, or (mean and std) will be used to create a normalizer.
+    If more than one set of inputs are given priority will be given following the order in which they are listed above.
+
+    Args:
+        file (str or Path): path to pt or npz file.
+        state_dict (dict): a state dict for Normalizer module
+        tensor (Tensor): a tensor with target values used to compute mean and std
+        mean (float | Tensor): mean of target data
+        std (float | Tensor): std of target data
+        data_loader (DataLoader): a data loader used to estimate mean and std
+        num_batches (int): the number of batches used. If not given all batches are used.
+
+    Returns:
+        Normalizer
+    """
+    # path takes priority if given
+    if file is not None:
+        if state_dict is not None or tensor is not None or mean is not None:
+            logging.warning(
+                "A file to a normalizer has been given. Normalization values will be read from it, and all other inputs"
+                " will be ignored."
+            )
+        extension = Path(file).suffix
+        if extension == ".pt":
+            # try to load a pt file
+            state_dict = torch.load(file)
+        elif extension == ".npz":
+            # try to load an NPZ file
+            values = np.load(file)
+            mean = values.get("mean")
+            std = values.get("std")
+            tensor = None  # set to None since values read from file are prioritized
+        else:
+            raise RuntimeError(
+                f"Normalizer file with extension '{extension}' is not supported."
+            )
+
+    # state dict is second priority
+    if state_dict is not None:
+        if tensor is not None or mean is not None:
+            logging.warning(
+                "The state_dict provided will be used to set normalization values. All other inputs will be ignored."
+            )
+        normalizer = Normalizer()
+        normalizer.load_state_dict(state_dict)
+        return normalizer
+
+    # if not then read target value tensor
+    if tensor is not None:
+        if mean is not None:
+            logging.warning(
+                "Normalization values will be computed from input tensor, all other inputs will be ignored."
+            )
+        mean = torch.mean(tensor)
+        std = torch.std(tensor)
```

Comment:
For forces (or potentially energy) we would want either the RMS force or the mean of the norm? How would that work?

Reply:
Thats a good point, and we need to make sure were are not using non-zero means to normalize forces.

Reply:
The configs are slightly different now and allow to override values. So to use force RMS instead of stdev and mean, you can just set the mean to zero like this,
```yml
      normalizer:
        fit:
          targets:
            forces: {mean: 0.0}
            energy: {}
```
All values that are explicitly set will override the fit.